### PR TITLE
Refactor auth

### PR DESCRIPTION
**Pull request by phamann**
- Remove current `auth` function
- Add `auth` lib to `create-email` function to validate in coming event data
- Add tests for `auth` lib
- Update cloudformation to pass `ft-user-uuid` and `ft-next-backend-key` to integration

### CODE REVIEW

File: functions/create-email/index.js [hunk 1]
```diff
@@ -82,7 +83,8 @@ exports.handle = λ((event) => {
 	console.log(event);
 	// If single request return HTML
 	if(!event.Records) {
-		return processEmail(event);
+		const authenticated = auth(event);
+		return authenticated ? processEmail(event) : Promise.reject('Unauthorized');
```
Comment (laurieboyes) [inline functions/create-email/index.js]: `Promise.reject` should be called with an error object

**Comment by laurieboyes** [comment]
Nice 👍  except minor comment

**Comment by phamann** [comment]
Have updated PR with tests

**Comment by laurieboyes** [comment]
Even nicer

**Pull request merged**